overlord/servicestate/quota_control.go

```diff
@@ -213,11 +214,20 @@ func ensureSnapServicesForGroup(st *state.State, grp *quota.Group, allGrps map[s
 		}
 	}
 
-	// now restart the services for each snap that was newly moved into a
-	// quota group
-	// TODO: handle snaps removed from a quota in #10218
+	// now restart the services for each snap that was newly moved into a quota
+	// group
 	nullPerfTimings := &timings.Timings{}
-	for sn, apps := range appsToRestartBySnap {
+	// iterate in a stable order over the snaps to restart their apps
+	snaps := make([]*snap.Info, 0, len(appsToRestartBySnap))
+	for sn := range appsToRestartBySnap {
+		snaps = append(snaps, sn)
+	}
+
+	sort.SliceStable(snaps, func(i, j int) bool {
```

Comment:
I'm a bit confused by the need for stable sort here, we get snap infos from map keys in no particular order and then just need them sorted by name, there is no other order we care about right? Should it be just sort.Slice (with the comment above adjusted)?

Reply:
You're right it doesn't need to be stable sorted, it just needs to be sorted. I generally prefer to use stable sorting when it's easy and we don't care about performance since it's more "natural" and produces less unexpected movements in lists, but sure we can just do sort.Slice() instead.